python/pyarrow/lib.pyx

```diff
@@ -143,6 +144,18 @@ def _pac():
     return __pac
 
 
+def _ensure_cuda_loaded():
+    # Try importing the cuda module to ensure libarrow_cuda gets loaded
+    # to register the CUDA device for the C Data Interface import
+    global __cuda_loaded
+    if not __cuda_loaded:
+        try:
+            import pyarrow.cuda  # no-cython-lint
+        except ImportError:
+            pass
```

Comment:
Do we really want to silence the error? The error message could be more informative than the error the user would later get when a device array fails importing...

Reply:
Good point. Pushed a change to capture the error message, and to actually raise a more informative error message here about pyarrow not being built with CUDA support (embedding the original import error message), instead of using the message from the C++ registry.